libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -105,7 +106,10 @@ bool StaticAnalyzer::visit(Return const& _return)
 bool StaticAnalyzer::visit(ExpressionStatement const& _statement)
 {
 	if (_statement.expression().annotation().isPure)
-		warning(_statement.location(), "Statement has no effect.");
+		m_errorReporter.warning(_statement.location(),
```

Comment:
My initial idea about this was to make all classes that have error reporting functionality _inherit_ from ErrorReporter. This way, changes to the code are smaller (for example, this line here would be unchanged), but still, we reduce code duplication.
Is this possible?